pkg/sql/plan/build_dml_util.go

```diff
@@ -979,7 +981,7 @@ func makeInsertPlan(
 	}
 
 	// make plan: sink_scan -> join -> filter	// check if pk is unique in rows & snapshot
-	if CNPrimaryCheck {
+	if CNPrimaryCheck && checkInsertPkDup {
```

Comment:
the checkInsertPkDup only used to check If we need to check if pk is unique in insert rows

can not used here.

when have table:   create table t1 (a int primary key, b int)

insert into t1 values (1,1);
the checkInsertPkDup is false.   because only insert one row.  we do not need to check dup for insert rows.
but we need to check dup between insert rows & rows in table.

Reply:
Fixed. Please verify.